core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -4315,6 +4315,192 @@ class KafkaApisTest extends Logging {
+            setPartitionIndex(0).
+            setCurrentLeaderEpoch(1)).asJava)).asJava)
+
+    val shareFetchRequest = new ShareFetchRequest.Builder(shareFetchRequestData).build(ApiKeys.SHARE_FETCH.latestVersion)
+    val request = buildRequest(shareFetchRequest)
+    kafkaApis = createKafkaApis()
+    kafkaApis.handleShareFetchRequest(request)
+    val response = verifyNoThrottling[ShareFetchResponse](request)
+    val responseData = response.data()
+
+    assertEquals(0, responseData.errorCode())
+    assertEquals(1, responseData.responses().size())
+    assertEquals(topicId, responseData.responses().get(0).topicId())
+    assertEquals(1, responseData.responses().get(0).partitions().size())
+    assertEquals(partitionIndex, responseData.responses().get(0).partitions().get(0).partitionIndex())
+    assertEquals(0, responseData.responses().get(0).partitions().get(0).errorCode())
+    assertEquals(0, responseData.responses().get(0).partitions().get(0).acknowledgeErrorCode())
+  }
+
+  @Test
+  def testHandleShareFetchRequestErrorPartitionsInCachedData(): Unit = {
+    val topicId = Uuid.randomUuid()
+    val partitionIndex = 0
+    addTopicToMetadataCache("foo", 1, topicId = topicId)
+    val memberId : Uuid = Uuid.ZERO_UUID
+
+    val topicIdPartition : TopicIdPartition = new TopicIdPartition(topicId, new TopicPartition(null, partitionIndex))
+    when(sharePartitionManager.newContext(any(), any(), any(), any())).thenReturn(
+      new SharePartitionManager.SessionlessShareFetchContext(Map(topicIdPartition -> new ShareFetchRequest.SharePartitionData(topicId, 1000, Optional.empty())).asJava)
+    )
+
+    when(clientQuotaManager.maybeRecordAndGetThrottleTimeMs(
+      any[RequestChannel.Request](), anyDouble, anyLong)).thenReturn(0)
+
+    val shareFetchRequestData = new ShareFetchRequestData().
+      setGroupId("group").
+      setMemberId(memberId.toString).
+      setTopics(List(new ShareFetchRequestData.FetchTopic().
+        setTopicId(topicId).
+        setPartitions(List(
+          new ShareFetchRequestData.FetchPartition().
+            setPartitionIndex(0).
+            setCurrentLeaderEpoch(1)).asJava)).asJava)
+
+    val shareFetchRequest = new ShareFetchRequest.Builder(shareFetchRequestData).build(ApiKeys.SHARE_FETCH.latestVersion)
+    val request = buildRequest(shareFetchRequest)
+    kafkaApis = createKafkaApis()
+    kafkaApis.handleShareFetchRequest(request)
+    val response = verifyNoThrottling[ShareFetchResponse](request)
+    val responseData = response.data()
+
+    assertEquals(0, responseData.errorCode())
+    assertEquals(1, responseData.responses().size())
+    assertEquals(topicId, responseData.responses().get(0).topicId())
+    assertEquals(1, responseData.responses().get(0).partitions().size())
+    assertEquals(partitionIndex, responseData.responses().get(0).partitions().get(0).partitionIndex())
+    assertEquals(100, responseData.responses().get(0).partitions().get(0).errorCode())
```

Comment:
I'm struggling to see where this 100 is coming from. Proper error constants would make it easier to follow again.

Reply:
I have added a short comment explaining this. If you see, name of the created topicIdPartition is null here. When this is discovered in the shareFetchRequest, UNKNOWN_TOPIC_ID is set as the error code for that partition